backend/eventloop/application_event_loop/application_event_runner_deployments.go

```diff
@@ -272,7 +271,36 @@ func (a applicationEventLoopRunner_Action) handleNewGitOpsDeplEvent(ctx context.
 		sourceRepoURL:        gitopsDeployment.Spec.Source.RepoURL,
 		sourcePath:           gitopsDeployment.Spec.Source.Path,
 		sourceTargetRevision: gitopsDeployment.Spec.Source.TargetRevision,
-		automated:            strings.EqualFold(gitopsDeployment.Spec.Type, managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated),
+		// syncOptions:       if non-empty, it gets updated below.
+		automated: strings.EqualFold(gitopsDeployment.Spec.Type, managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated),
+	}
+
+	if gitopsDeployment.Spec.SyncPolicy != nil && len(gitopsDeployment.Spec.SyncPolicy.SyncOptions) != 0 {
+		// syncOption = gitopsDeployment.Spec.SyncPolicy.SyncOptions
+		for _, syncOptionString := range gitopsDeployment.Spec.SyncPolicy.SyncOptions {
+			// Checks for each SyncOption goes in this for loop
+			var checkSyncOption bool
+			// 1. Check for SyncOption "CreateNamespace=true" / empty string
+			if syncOptionString == "" {
+				checkSyncOption = true
+			}
+			if syncOptionString == "- CreateNamespace=true" {
```

Comment:
I think in a GitOpsDeployment CR the SyncOption will be added exactly the way it is done in ArgoCD resource i.e with a "- " 

```
apiVersion: managed-gitops.redhat.com/v1alpha1
kind: GitOpsDeployment

metadata: 
  name: my-gitops-depl
  namespace: jane

spec: 

  source: 
    repoURL: https://github.com/redhat-appstudio/managed-gitops
    path: resources/test-data/sample-gitops-repository/environments/overlays/dev

  destination: 
    namespace: jane

  # maintain Argo CD API compatibility, by adding these 3 new fields
  syncPolicy: 
    syncOptions: 
      - CreateNamespace=true
```

because of which I replicated the same string which will eventually get sanitised (in backend component) before being inserted in database and finally in the applicationCR

Reply:
As discussed in standup, the "- " should not be included.  The dash is actually the way in yaml to indicate an entry is an item in a list.